chubby-overhaul-mute

CirnoMoment opened the pull request:
- Created Temp-Mute Command w/ `roles.muted` setting.
- Added MS Module, which allows certain times to be set, ranging from milliseconds to years.
- Allows Discord's Timeout feature to be used side by side.

jonbarrow commented on src/commands/mute.js:
```diff
@@ -0,0 +1,138 @@
+
+        setTimeout(function(){
+            member.roles.remove(mutedRole);
+        }, ms(timeamount));
```
Whenever the bot restarts these timeouts will be lost. There needs to be a recovery system when the bot boots to go through the DB and check for any mutes that have not expired and re-make their timeouts, and to unmute anyone who's mute expired while the bot was offline

jonbarrow commented on src/commands/mute.js:
```diff
@@ -0,0 +1,138 @@
+
+		await member.send({
+			embeds: sendMemberEmbeds
+		}).catch(() => console.log('Failed to DM user'));
```
The formatting for these lines is off, it's indented too far and doesn't line up

jonbarrow commented on src/commands/mute.js:
```diff
@@ -0,0 +1,138 @@
+			timeamount: timeamount
+		});
+
+		muteListEmbed.addField(`${member.user.username}'s mute time is `, timeamount.toString(), true);
```
The formatting for these lines is off, it's indented too far and doesn't line up

jonbarrow commented on src/commands/mute.js:
```diff
@@ -0,0 +1,138 @@
+	.addStringOption(option => {
+		return option.setName('time')
+			.setDescription('Time of the mute (ex: 15s, 2 hrs, 14 days)')
+			.setRequired(true);
```
Maybe instead of requiring a time to be set, make it optional and if no time is given then it's a permanent mute?

jonbarrow commented on src/events/guildMemberUpdate.js:
```diff
@@ -20,6 +21,11 @@ async function guildMemberUpdateHandler(oldMember, newMember) {
+		limit: 1,
+		type: 'MEMBER_ROLE_UPDATE'
+	})) return;
+
```
What is the point of this here? It's not clear what this fixes. Also this `if` statement is formatted in a very non-readable way

CirnoMoment replied:
I have updated this into one line, making it a skip instead of an entire check function.

jonbarrow commented on src/models/mutes.js:
```diff
@@ -0,0 +1,25 @@
+		type: DataTypes.DATE,
+		defaultValue: DataTypes.NOW
+	}
+});
```
The formatting for these lines is off, the indentation doesn't line up

jonbarrow reviewed (changes requested):
Only a few styling changes with minor feature tweaks

CirnoMoment commented:
Alright added some of your suggestions and fixes:
- Seems like formatting is solely on Github web's end as the formatting on my end _(even on Github Desktop)_ isn't as indented (see image below).
![ScreenShot_10_18_2022_5_08_45_PM](https://user-images.githubusercontent.com/111249711/196577835-568a375a-ed1c-484b-8494-f11154ff470e.png)
- Added explanation to the if statement in `GuildMemberUpdate.js`, which fixes an error that I found during testing where it would call said js file when muting/unmuting.
- Added the option to perm/temp mute as requested, also added an unmute command aswell.
- I cannot figure out for the life of me how to convert the mute into the database, so unfortunately I couldn't add that fix. However, if you want, I have allowed collab with you just if you want to add that feature.

jonbarrow commented:
> Seems like formatting is solely on Github web's end as the formatting on my end (even on Github Desktop) isn't as indented (see image below).

No it's on your end. Are you following the linting rules? The issue is that your code has a mix of spaces and tabs used for indentation. This might look the same in your IDE but GitHub shows the formatting differently. We use tabs with a width of 4, it seems like you are using spaces. You can clearly see the use of spaces in the following image (the first `if` is indented with spaces, while the second one is indented with tabs)
![code](https://i.imgur.com/LQhXJkx.png)

> Added explanation to the if statement in GuildMemberUpdate.js, which fixes an error that I found during testing where it would call said js file when muting/unmuting.

I see you added the comment, but it did not address the fact that the check/return is very unreadable as it is

> I cannot figure out for the life of me how to convert the mute into the database, so unfortunately I couldn't add that fix. However, if you want, I have allowed collab with you just if you want to add that feature.

Store `currentTime + ms(timeout)` in the DB with the warn to know when it expires. When the bot boots up again on the `ready` event, pull all warns from the DB and loop over them. Check if the expired timestamp is older than the current time. If it is, then unmute the user immediately. If not, make a new timeout using `expireTime - currentTime` to get the time remaining until their mute should be cleared

Also I opened a review with review comments, please reply directly to the comments during review so it can be properly tracked when each review is resolved

CirnoMoment commented:
Hm, that's odd, so I looked through my code again, and all of it seems to be indented using the tab button, which the main repo seems to be as well, but the issue of it being too far indented is reflected in main branch too, as seem here:
![ScreenShot_10_20_2022_4_31_56_PM](https://user-images.githubusercontent.com/111249711/197062949-f4c81b68-acff-433b-aef8-e957a5de2e79.png)
And I saw that I did have a function using only spaces within the code, but that was this line, where you didn't mark down for review:
![ScreenShot_10_20_2022_4_30_07_PM](https://user-images.githubusercontent.com/111249711/197063226-65ace534-6222-4801-94e1-cb3d258cdb5d.png)
Outside of that, I'll try make that one check more readable for others, and I'll try working on that offline temp mute check too, although I'm already getting into issues within my testing for that. Once I get it working (hopefully) I'll comment on the review to track the change.

CirnoMoment commented:
Alright, so I think I found a solution for the storing it into a database, but I just need some clarification on some things. So when the `setTimeout` function is called, it will create a new line within a list that will be called "mute-list". Each line within this list will store 3 variables, that being the member, the time when it was carried, and the amount of time said user is muted. The only reason why I'm asking is that is that I do not know how to create a list/add lines within the list using the db.js functions. I think I know what is needed to be done, but I just need some help making sure the list is set up correctly.
![ScreenShot_10_20_2022_6_28_42_PM](https://user-images.githubusercontent.com/111249711/197078570-64c97eea-179f-4073-94c3-8450fa3b92d9.png)
(Image above is what I have for storing data, feel free to correct me as I'm fairly new to databases.)

CirnoMoment commented:
Alright, if you haven't noticed from my other commits from a week ago, I removed the spacing and added some minor adjustments, so please feel free to review that stuff. Also, I've tried several times to get that DB working, but my bad js skills are showing 😓. Again, I feel guilty, but maybe you could add the DB? It should be as easy as just simply adding a loop and an if else statement within `ready.js` and setting that said data within the timeout function in `mute.js`. The issue, as seen previously, is that I do not understand db and lists that much. I've gave you collab access to add it once you're not busy, again sorry for asking you to do this when I'm supposed to do it.
